connmgr.go

```diff
@@ -34,16 +35,17 @@ type BasicConnMgr struct {
 	protected map[peer.ID]map[string]struct{}
 
 	// channel-based semaphore that enforces only a single trim is in progress
-	trimRunningCh chan struct{}
-	trimTrigger   chan chan<- struct{}
-	connCount     int32
+	trimMutex sync.Mutex
+	connCount int32
+	trimCount uint64 // to be accessed atomically. This is mimicking the implementation of a sync.Once.
```

Comment:
this is potentially unaligned... It either needs to be int32 or we have to use a pointer.

Reply:
It was unaligned, but it's not any more after I rearranged it. We're testing on 32 and 64 bit on CI, so we don't have to worry about alignment, as long as tests pass.

Reply:
can we add a comment regarding alignment to aid future selves?

Also, it is aligned now in 32bit but off by 4 in 64 bit which may cause issues in non intel processors! Maybe we should just put it first?

Reply:
Added a comment.